File: test_resources/rust/struct_with_parameters_models.rs
#[derive(Debug, Clone, PartialEq)]
pub struct Test {
    pub delta_time: f32,
    pub view_width: f32,
    pub view_height: f32,
    pub touch_start_x: f32,
    pub touch_start_y: f32,
    pub last_touch_x: f32,
    pub last_touch_y: f32,
    pub touch_x: f32,
    pub touch_y: f32,
}

impl Default for Test {
    fn default() -> Self {
        Self {
            delta_time: 0.0,
            view_width: 0.0,
            view_height: 0.0,
            touch_start_x: 0.0,
            touch_start_y: 0.0,
            last_touch_x: 0.0,
            last_touch_y: 0.0,
            touch_x: 0.0,
            touch_y: 0.0,
        }
    }
}

File: test_resources/dart/struct_with_parameters_models.dart
class Test {
  double deltaTime;
  double viewWidth;
  double viewHeight;
  double touchStartX;
  double touchStartY;
  double lastTouchX;
  double lastTouchY;
  double touchX;
  double touchY;

  Test({
    required this.deltaTime,
    required this.viewWidth,
    required this.viewHeight,
    required this.touchStartX,
    required this.touchStartY,
    required this.lastTouchX,
    required this.lastTouchY,
    required this.touchX,
    required this.touchY,
  });

  Test.createDefault()
      : deltaTime = 0.0,
        viewWidth = 0.0,
        viewHeight = 0.0,
        touchStartX = 0.0,
        touchStartY = 0.0,
        lastTouchX = 0.0,
        lastTouchY = 0.0,
        touchX = 0.0,
        touchY = 0.0;
}

class TestBuffersFactory implements BuffersFactory<Test> {
  const TestBuffersFactory();

  @override
  Test createDefault() => Test.createDefault();
}
